diffpy/snmf/datapoint.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import numdifftools
+
+
+class ComponentSignal:
+    """
+    Attributes
+    ----------
+    iq: 1d array of floats
+      The intensity/g(r) values of the component
+    grid: 1d array of floats
+      The vector containing the grid points of the component
+    weights: 1d array of floats
+      The vector containing the weight of the component signal for each signal
+    stretching_factors: 1d array of floats
+      The vector containing the stretching factor for the component signal for each signal
+    id: int
+      The component number.
+    """
+    def __init__(self, grid, iq, weights, stretching_factors, id_number):
+        self.iq = np.asarray(iq)
+        self.grid = np.asarray(grid)
+        self.weights = np.asarray(weights)
+        self.stretching_factors = np.asarray(stretching_factors)
+        self.id = int(id_number)
+        self.stretched_iqs = ()
+
+    def apply_stretch(self, m):
+        """Applies a stretching factor to a component
+
+        Parameters
+        ----------
+        m: int
+          The index specifying which stretching factor to apply
+
+        Returns
+        -------
+        tuple of 1d arrays
+          The tuple of vectors where one vector is the stretched component, one vector is the gradient of the stretching
+          operation, and one vector is the second derivative of the stretching operation.
+
+        """
+        normalized_grid = np.arange(len(self.grid))
```

Comment:
Is there a reason you don't  just initialize normalized grid as a copy of self.grid `normalized_grid = copy(self.grid)`?   I don't really understand what is happening, but I would worry that the derivative would look different if you stretch a grid whose values are something like 100x different than the actual grid.

Reply:
I believe you're right. I do this because the matlab code does this. I asked Ran the same question earlier, and he said that it shouldn't matter as long as the points are equally spaced.